app/models/project.rb

```diff
@@ -131,16 +132,22 @@ def violates_location?(obs)
     !location.found_here?(obs)
   end
 
+  def violates_dates?(obs)
+    dates_exclude?(obs.when)
+  end
+
   def count_violations
-    return 0 unless location
+    return out_of_range_observations.count unless location
 
-    count = observations.where.not(lat: nil).count
-    count - observations.in_box(n: location.north, s: location.south,
-                                e: location.east, w: location.west).count
+    count =
+      out_of_range_observations.or(observations.where.not(lat: nil)).count
+    count - in_range_observations.
+            in_box(n: location.north, s: location.south,
+                   e: location.east, w: location.west).count
   end
```

Comment:
@mo-nathan 
Would you mind double-checking the logic of the revised `count_violations`.

Reply:
The changes seem right, but there's a bug in the original implementation.  It doesn't handle cases where observations have a location, but no lat/long.  Essentially it is missing the logic reflected in line 280 from location.rb (last line of found_here?).